core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -909,8 +911,35 @@ private[group] class GroupCoordinator(
         val group = groupManager.getGroup(groupId).getOrElse {
           groupManager.addGroup(new GroupMetadata(groupId, Empty, time))
         }
-        doTxnCommitOffsets(group, transactionalId, memberId, groupInstanceId, generationId, producerId, producerEpoch,
-          offsetMetadata, requestLocal, responseCallback)
+
+        val offsetTopicPartition = new TopicPartition(Topic.GROUP_METADATA_TOPIC_NAME, partitionFor(group.groupId))
+
+        def postVerificationCallback(
+          error: Errors,
+          newRequestLocal: RequestLocal,
+          verificationGuard: VerificationGuard
+        ): Unit = {
+          if (error != Errors.NONE) {
+            val finalError = error match {
+              case Errors.NOT_ENOUGH_REPLICAS => Errors.COORDINATOR_NOT_AVAILABLE
+              case e => e
+            }
+            responseCallback(offsetMetadata.map { case (k, _) => k -> finalError })
+          } else {
+            doTxnCommitOffsets(group, memberId, groupInstanceId, generationId, producerId, producerEpoch,
+              offsetTopicPartition, offsetMetadata, newRequestLocal, responseCallback, Some(verificationGuard))
+          }
+        }
+
+        groupManager.replicaManager.maybeStartTransactionVerificationForPartition(
```

Comment:
The access to `replicaManager` here probably suggests that we probably should be going through `GroupMetadataManager`. We could expose a wrapped `maybeStartTransactionVerificationForPartition` from `GroupMetadataManager` instead. That might also help us encapsulate the error conversion a little better.

Reply:
i was just about to push my change before I saw this comment. I will address this comment tomorrow.